vllm/config.py

```diff
@@ -271,9 +278,20 @@ def _init_pooler_config(
         pooling_returned_token_ids: Optional[List[int]] = None
     ) -> Optional["PoolerConfig"]:
         if self.task == "embedding":
+            pooling_type_ = pooling_type
+            normalize_ = pooling_norm
+            pooling_config = get_pooling_config(self.model, self.revision)
+            if pooling_config is not None:
+                pooling_type_ = pooling_config["pooling_type"]
+                normalize_ = pooling_config["normalize"]
+                # override if user specifies pooling_type and/or pooling_norm
+                if pooling_type is not None:
+                    pooling_type_ = pooling_type
+                if pooling_norm is not None:
+                    normalize_ = pooling_norm
```

Comment:
```suggestion
            pooling_config = get_pooling_config(self.model, self.revision)
            if pooling_config is not None:
                # override if user specifies pooling_type and/or pooling_norm
                if pooling_config["pooling_type"] is not None:
                    pooling_type = pooling_config["pooling_type"]
                if pooling_config["normalize"]  is not None:
                    normalize = pooling_config["normalize"]
```

Reply:
This way, the pooling config from model's files will override the user input. Is that the expected behavior for this?

Reply:
Oops, let me edit this.

Reply:
Edited.

Reply:
Got it! Thank you!